hax/test/integration/test_motr.py

```diff
@@ -177,3 +242,208 @@ def my_services(name):
+
+
+def test_broadcast_node_failure(
+        mocker, planner, motr, consumer, consul_util):
+    def new_kv(key: str, val: str):
+        return {
+            'Key': key,
+            'CreateIndex': 1793,
+            'ModifyIndex': 1793,
+            'LockIndex': 0,
+            'Flags': 0,
+            'Value': val,
+            'Session': ''
+        }
+
+    def my_get(key: str, recurse: bool = False):
+        if key == 'm0conf/nodes' and recurse:
+            return [new_kv(k, v) for k, v in [(
+                'm0conf/nodes/0x6e00000000000001:0x3/processes'
+                '/0x7200000000000001:0x15',
+                '{\"name\": \"m0_server\", \"state\": \"offline\"}'), (
+                    'm0conf/nodes/cmu/processes/6/services/rm', '16'),
+                ('m0conf/nodes/localhost/processes/7/services/rms',
+                    '17'),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0x15/services/0x7300000000000001:0x17',
+                 '{\"name\": \"ios\", \"state\": \"failed\"}'),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0xa/services/0x7300000000000001:0xc',
+                 '{\"name\": \"ios\", \"state\": \"failed\"}')]
+            ]
+        elif key == 'm0conf/sites' and recurse:
+            return [new_kv(k, v) for k, v in [(
+                'm0conf/sites/0x5300000000000001:0x1/racks'
+                '/0x6100000000000001:0x2/encls/0x6500000000000001:0x4'
+                '/ctrls/0x6300000000000001:0x5',
+                '{\"state\": \"M0_NC_UNKNOWN\"}'),
+                ('m0conf/sites/0x5300000000000001:0x1/racks'
+                 '/0x6100000000000001:0x2/encls/0x6500000000000001:0x4'
+                 '/ctrls/0x6300000000000001:0x6',
+                 '{\"state\": \"M0_NC_UNKNOWN\"}')]
+            ]
+        elif (key == 'm0conf/nodes/0x6e00000000000001:0x3'
+              '/processes' and recurse):
+            return [new_kv(k, v) for k, v in [(
+                'm0conf/nodes/0x6e00000000000001:0x3/processes'
+                '/0x7200000000000001:0x15',
+                '{\"name\": \"m0_server\", \"state\": \"failed\"}'),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0x15/services/0x7300000000000001:0x17',
+                 '{\"name\": \"ios\", \"state\": \"failed\"}'),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0xa/services/0x7300000000000001:0xc',
+                 '{\"name\": \"ios\", \"state\": \"failed\"}')]
+            ]
+        elif (key == 'm0conf/nodes/0x6e00000000000001:0x3/processes'
+              '/0x7200000000000001:0x15' and recurse):
+            return [new_kv(k, v) for k, v in [(
+                'm0conf/nodes/0x6e00000000000001:0x3/processes'
+                '/0x7200000000000001:0x15',
+                '{\"name\": \"m0_server\", \"state\": \"failed\"}'),
```

Comment:
[Optional]: Instead of playing with escaping characters, one could use something like `json.dumps({'name': 'm0_server', 'state': 'failed'}). It is a bit more readable and such code can be formatted automatically by the formatter tool (while long strings are not that easy to format).

Reply:
Done